examples/contrib/oed/gp_bayes_opt.py

```diff
@@ -0,0 +1,164 @@
+# import matplotlib.gridspec as gridspec
+# import matplotlib.pyplot as plt
+import torch
+import torch.autograd as autograd
+import torch.optim as optim
+from torch.distributions import transform_to
+
+
+class GPBayesOptimizer:
```

Comment:
We should be able to refactor this a bit into a `pyro.optim.multi.MultiOptimizer` so that it looks more like other Pyro optimizers.  That way when we experiment with other optimizers e.g. gradient-based optimizers it'll be easy to swap them out

Reply:
I think we're best to talk over this together- we need to implement a `step` function which would replace my `run` function. Would be also need to refactor the example to use `pyro.param`? Right now, it's not really true that after a `step`, the params are updated to new near-optimal values. But obviously I could add `self.opt_differentiable(lambda x: self.gpmodel(x)[0])` into each step (minimize the current GP mean function)